stdlib/InteractiveUtils/src/codeview.jl

```diff
@@ -59,9 +59,13 @@ See [`@code_warntype`](@ref man-code-warntype) for more information.
 """
 function code_warntype(io::IO, @nospecialize(f), @nospecialize(t=Base.default_tt(f));
                        debuginfo::Symbol=:default, optimize::Bool=false, kwargs...)
+    code_warntype(io, code_typed(f, t; optimize, kwargs...))
+end
+
+function code_warntype(io::IO, code_typed_output::Vector{Any})
```

Comment:
I think it makes more sense to rename this function to something like `show_code_warntype` and work over a single `(src, rettype)` at a time.